### PR TITLE
Config metadata

### PR DESCRIPTION
**Pull request by hcorson-dosch**
* Added attribute definitions for `pgdl_fits_N***_W***.zip`, `pgdl_config.csv` and `pgdl_fit_lakes.png`
* Added Python build environment for pgdl ( `lakes_pgdl` conda environment)

Per the list of things @lindsayplatt  came up with that she checked during my update of and additions to text_07_habitat.yml metadata ( in #27 ):

- [x] ~Update the basics as needed: title, abstract, purpose, start/end date (of data), themekeywords, process-date, indirect-spatial, latitude-res, longitude-rest, data-name, data-description, file-format (see mntoha examples and hyperscales examples)~
 - [x] ~Remove the entries that only need to exist in the shared text file (authors, pubdate, doi, contacts (incl metadata contact), accuracy test, funding creds, process description, distribution person, liability statement)~ @lindsayplatt looked and didn't see any of this information in text_03_config.yml as of now.
- [x]  Add entities for each file (Lindsay "stubbed" in the 3 additional entities and attributes that will be needed in #28)
- [x]  Add attributes for contents of each file
  - Need attribute definition for `phase` in `pgdl_config.csv` @aappling-usgs how would you define pretrain vs finetune? I wasn't certain what prior knowledge we can assume, or how best to succinctly describe the two different phases.
  - @aappling-usgs - if you could review my attribute definitions for `goal` and `fold`, as well as my edits to the `data-description` at line 335, that would be great
- [x]  Include appropriate citations ("larger citations" and cross citations)
  - Looks like Jordan's 2019 paper is cited, @jread-usgs, @aappling-usgs, does any other work need to be cited here?
- [x]  Add build environment for PGDL (GLM is complete)
  - Added Python build environment for pgdl, but not yet R build environment. @lindsayplatt or @jread-usgs is there a way to extract the versions for all packages in a project's Rlib folder? 
- [x]  Build the XML using scmake("out_xml/[STEP]_sb.xml")
- [x]  Validate the XML output using https://mrdata.usgs.gov/validation/
- [ ]  Push to SB using scmake("log/[STEP]_sb_data.csv") (note you will need to have built the data targets listed as arguments in the remake.yml

### CODE REVIEW

**Comment by lindsayplatt** [comment]
There is a way to print out all R packages and their versions. I did it using `remake::dump_environment()` (as described in #24) but @jread-usgs used something else more robust I think? I was looking for what it was but I can't find notes anywhere about it.

**Comment by jread-usgs** [comment]
I use `meddle::environment_metadata()`

File: in_text/text_03_config.yml [hunk 1]
```diff
@@ -180,43 +180,129 @@ entities:
       data-units: NA
   -
     data-name: pgdl_fits_{group_id}_N{minN-maxN}_W{minW-maxW}.zip
-    data-description: see this example https://github.com/USGS-R/mntoha-data-release/blob/master/in_text/text_07_habitat.yml#L117-L121
+    data-description: A zip file containing a folder for every lake within {group_id} with the files needed to re-run the trained model with tensorflow.
     attributes:
-      -
```
Comment (jread-usgs) [inline in_text/text_03_config.yml:185]: Are there not any attributes in these files? (e.g., files with columns?)
I can go look at the files and provide some advice, but I think you need to include some attributes for an entity.

File: in_text/text_03_config.yml [hunk 2]
```diff
@@ -180,43 +180,129 @@ entities:
...
+        data-units: daily observations
+      -
+        attr-label: max_batch_obs
+        attr-def: Upper limit on number of individual temperature predictions (date-depth-split combos) to generate and test against observations or internal consistency per batch. Used to decide how many batches to run. True batch size will differ between supervised and unsupervised datasets and will be capped at (1) for the larger of supervised and unsupervised, the largest number of completes sequences for complete depth profiles that fit within this max_batch_size, and (2) for the smaller of supervised adn unsupervised, the number of complete sequences that will make an equal number of batches to the other dataset type.
+        attr-defs: This data release
+        data-min: 200000
+        data-max: 200000
+        data-units: number of individual temperature predictions
+      -
+        attr-label: restore_path
```
Comment (jread-usgs) [inline in_text/text_03_config.yml:290]: wondering if `restore_path` and `save_path` are relevant to export or should be removed?

File: in_text/text_03_config.yml (hunk 1 above)
Comment (hcorson-dosch) [reply in_text/text_03_config.yml:185]: They are tensorflow files.:
image

The compressed numpy file `model_config.npz` contains the tensorflow graph config.

@aappling-usgs and I chatted a bit about what level of detail to include in the metadata. Since we aren't including the code to generated new predictions with these trained model configurations, I believe Alison was thinking we could keep it brief for now.

Comment (hcorson-dosch) [reply in_text/text_03_config.yml:185]: None of those files are human-readable through a text editor or anything.

Much of the information stored within the graph config aligns with that defined in `pgdl_config.csv`:
image

**Comment by hcorson-dosch** [comment]
Note - added pgdl R environment earlier today

**Review by aappling-usgs — APPROVED** [review]
Suggestions for `phase`, `goal`, `fold`, and `data-description` are inline.

> Looks like Jordan's 2019 paper is cited, does any other work need to be cited here?
I think that's sufficient for citations.

File: in_text/text_03_config.yml (hunk 2 above)
Comment (hcorson-dosch) [reply in_text/text_03_config.yml]: removed from metadata and `pgdl_config.csv` in latest commits. Also pushed updated `pgdl_config.csv` to SB. Note that `pgdl_fits_XX_NXXX_WXXX.zip` files and `pgdl_fit_lakes.png` also got updated, even though they haven't been rebuilt since December

File: in_text/text_03_config.yml (hunk 1 above)
Comment (hcorson-dosch) [reply in_text/text_03_config.yml:185]: I built and validated the xml, and it didn't throw any error for not having any attributes defined for `pgd_fits_{group_id}_N{minN-maxN}_W{minW-maxW}.zip`:

image

So I think it's up to us whether or not we wish to provide one "attribute" per file within the zipfile, per your suggestion, @aappling-usgs. My only hesitation is that I don't know how we'd define the 3 files that aren't the graph config.

Thoughts @aappling-usgs and @jread-usgs ?

**Review by jread-usgs — APPROVED** [review]
:+1:
